### PR TITLE
DX: remove `@group` annotations

### PR DESCRIPTION
**Pull request by kubawerlos**
We define tests groups via testsuites, so we don't need `@group` annotation anymore, do we?

### CODE REVIEW

File: .php-cs-fixer.dist.php [hunk 1]
```diff
@@ -37,7 +37,10 @@
         '@PHPUnit100Migration:risky' => true,
         '@PhpCsFixer' => true,
         '@PhpCsFixer:risky' => true,
-        'general_phpdoc_annotation_remove' => ['annotations' => ['expectedDeprecation']], // one should use PHPUnit built-in method instead
+        'general_phpdoc_annotation_remove' => ['annotations' => [
+            'expectedDeprecation', // one should use PHPUnit built-in method instead
+            'group',
```
Comment (keradus) [inline .php-cs-fixer.dist.php:42]: as we already have some reasons here for the non-obvious removal of annotation, would be good to continue with it for new annotation, so when JohnDoe open this file in future, they do know the reason.

File: tests/TextDiffTest.php [hunk 2]
```diff
@@ -24,8 +24,6 @@
  * @internal
  *
  * @coversNothing
- *
- * @group covers-nothing
```
Comment (keradus) [inline tests/TextDiffTest.php:28]: i'm not convinced. Isn't it changing the logic?
like, for coverage job, on master we do not test this file, but on your PR we do?

**Review by Wirone — COMMENTED** [review]
I am not sure if it's a good direction. Suites and groups are not redundant per se, they complement each other. Yes, at this point `@group auto-review` does not make sense because there's a suite for that, or `@group legacy` because it was required for Symfony's bridge which was removed, but IMHO it does _not_ make sense to prevent `@group` usage and to remove _all_ of them using configured fixer. Groups are for cherry-picking subset of tests from the suite(s) and losing that ability on our own wish is wrong, from my perspective. Without `@group` the only way of narrowing the scope would be passing paths to the command, but it would only work for tests grouped physically under the same directory, not spread across several directories. So I think we should clean up obsolete annotations, but keep the ability to use `@group` when it makes sense.

File: tests/TextDiffTest.php (hunk 2 above)
Comment (Wirone) [reply tests/TextDiffTest.php:28]: Exactly, it's an example of what I said [here](https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/pull/7611#pullrequestreview-1794057188).

Comment (kubawerlos) [reply tests/TextDiffTest.php:28]: > for coverage job, on master we do not test this file, but on your PR we do?

why would you want to run this file in a coverage job?

Comment (keradus) [reply tests/TextDiffTest.php:28]: I would not like to. but if I read this PR properly, on your PR this file is getting executed in CI-code-coverage job

Comment (kubawerlos) [reply tests/TextDiffTest.php:28]: > on your PR this file is getting executed in CI-code-coverage job

how did you come to this conclusion? How do you read the changes in `phpunit.xml.dist` then?

Comment (keradus) [reply tests/TextDiffTest.php:28]: looks like I overlooked changes in that file - not the best from me, multiple files 36 files affected overloaded me, apparently ;)

but hey, that's exactly why I asked "isn't it changing logic?" or "(what) on your PR we do?" and not "this is wrong, go away".

Comment (kubawerlos) [reply tests/TextDiffTest.php:28]: I've opened this PR with, I thought, quite a straightforward change and clear description, and got two answers starting with "i'm not convinced" and "I am not sure if it's a good direction". Both are not encouraging, don't you agree?

Comment (keradus) [reply tests/TextDiffTest.php:28]: One person asking for help to understand "how this PR works?", while other was wondering if it makes sense to forbid whole feature.
They serve different purpose, but not of them is harshly pushing "go away", but opening the doors for better understanding or alignment.

**Pull request closed without merging**